tests/e2e_logging.py

```diff
@@ -1202,17 +1204,31 @@ def test_receipts(network, args):
 @reqs.at_least_n_nodes(2)
 def test_random_receipts(network, args):
     primary, _ = network.find_primary_and_any_backup()
-    cert_path = os.path.join(primary.common_dir, f"{primary.local_node_id}.pem")
-    with open(cert_path) as c:
-        node_cert = load_pem_x509_certificate(
-            c.read().encode("ascii"), default_backend()
-        )
+
+    common = os.listdir(network.common_dir)
+    cert_paths = [
+        os.path.join(network.common_dir, path)
+        for path in common
+        if re.compile(r"^\d+\.pem$").match(path)
+    ]
+    certs = {}
+    for path in cert_paths:
+        with open(path) as c:
+            cert = c.read()
+        certs[
+            infra.crypto.compute_public_key_der_hash_hex_from_pem(cert)
+        ] = load_pem_x509_certificate(cert.encode("ascii"), default_backend())
 
     with primary.client("user0") as c:
         r = c.get("/app/commit")
-        max_view, seqno = [int(e) for e in r.body.json()["transaction_id"].split(".")]
+        max_view, max_seqno = [
+            int(e) for e in r.body.json()["transaction_id"].split(".")
+        ]
         view = 2
-        for s in sorted(random.sample(range(1, seqno), min(50, seqno))) + [seqno]:
+        seqnos = range(3, max_seqno)
+        for s in (
+            [1, 2] + sorted(random.sample(seqnos, min(50, len(seqnos)))) + [max_seqno]
```

Comment:
there are some magic numbers here, if at all possible can they become variable/constants

Reply:
Yes, I will do that. Just to explain the rationale here, 1 and 2 are to cover genesis and probably the first signature (it is today, but not guaranteed), because they're always a little special. The last one is to cover at least a signature for sure, since it's a commit point we know it will be.

The rest is just a random sampling, to keep runtime in check.